### PR TITLE
Feature/128 add application pagescreens

### PR DESCRIPTION
**Pull request by RohitGarudadri**
### Tickets:

HCK 128

### List of changes:

Made the application form a multipart form for users through work though. General questions -> short answer questions -> accommodation questions -> other questions

### Type of change:

Please delete options that aren't relevant.
- [x] New feature (non-breaking change which adds functionality)

### How did you do this?

Refactored the application container to render 4 different forms dependent on the page number the user is on. Only allowed forward movement if the current section was valid. Free movement backwards though.

### Why are you choosing this approach?

Only approach?

### PR Checklist:

- [x] Merged `develop` branch (before testing)
- [x] Linted my code locally
- [x] Listed change(s) in the Changelog
- [x] Tested all links in project relevant browsers
- [x] Tested all links on different screen sizes
- [x] Referenced all useful info (issues, tasks, etc)

### CODE REVIEW

File: src/features/Application/ManageApplicationContainer.tsx [hunk 1]
```diff
@@ -130,49 +145,57 @@ class ManageApplicationContainer extends React.Component<
...
+        <MaxWidthBox m={'auto'} maxWidth={'500px'}>
+          <Sidebar
+            currentPage="Application"
+            status={this.state.hackerDetails.status}
+            confirmed={true}
+          />
+          <Helmet>
+            <title>
+              {mode === ManageApplicationModes.CREATE ? 'Create' : 'Edit'}
+              Application | McHacks 6
```
Comment (krubenok) [inline src/features/Application/ManageApplicationContainer.tsx]: Can the page title here be abstracted to the constants file? i.e. 

`Application | {CONSTANTS.HACKATHON_NAME}`

File: src/features/Application/ManageApplicationContainer.tsx [hunk 2]
```diff
@@ -364,66 +348,221 @@ class ManageApplicationContainer extends React.Component<
...
+        <ErrorMessage
+          name={'hacker.application.accommodation.shirtSize'}
+          component={FormikElements.Error}
+        />
+        {/* This fixes the issue with going back somehow, so leave it here temporarily */}
+        <div />
+        <FastField
+          name={'hacker.application.accommodation.impairments'}
+          component={FormikElements.LongTextInput}
+          label={'Do you have any impairments that we should know about?'}
```
Comment (krubenok) [inline src/features/Application/ManageApplicationContainer.tsx]: Same with this, most of the other labels have been abstracted to the constants file.

File: src/features/Application/ManageApplicationContainer.tsx [hunk 3]
```diff
@@ -364,66 +348,221 @@ class ManageApplicationContainer extends React.Component<
...
+        />
+        <ErrorMessage
+          component={FormikElements.Error}
+          name={'hacker.application.accommodation.impairments'}
         />
         <FastField
-          name={'application.other.codeOfConduct'}
+          name={'hacker.application.accommodation.barriers'}
+          component={FormikElements.LongTextInput}
+          label={'Do you have any barriers that we should know about?'}
```
Comment (krubenok) [inline src/features/Application/ManageApplicationContainer.tsx]: Ditto

File: src/features/Application/ManageApplicationContainer.tsx [hunk 4]
```diff
@@ -364,66 +348,221 @@ class ManageApplicationContainer extends React.Component<
...
+          label={CONSTANTS.BUS_REQUEST_LABEL}
+          subtitle={CONSTANTS.BUS_REQUEST_SUBTITLE}
+          required={false}
+        />
+        {fp.values.needsBus ? (
+          <React.Fragment>
+            <FastField
+              name={'hacker.application.accommodation.travel'}
+              component={FormikElements.FormattedNumber}
+              label="How much will you need to be reimbursed? (Up to $100)"
```
Comment (krubenok) [inline src/features/Application/ManageApplicationContainer.tsx]: Ditto

File: src/features/Application/ManageApplicationContainer.tsx [hunk 5]
```diff
@@ -181,177 +204,138 @@ class ManageApplicationContainer extends React.Component<
...
         />
         <ErrorMessage
           component={FormikElements.Error}
-          name="application.general.degree"
+          name="hacker.application.general.degree"
         />
         <FastField
-          name={'application.general.graduationYear'}
+          name={'hacker.application.general.graduationYear'}
           label="Graduation year:"
```
Comment (krubenok) [inline src/features/Application/ManageApplicationContainer.tsx]: Ditto

File: src/features/Application/validationSchema.ts [hunk 6]
```diff
@@ -1,80 +1,262 @@
...
+          application: object().shape({
+            general: object().shape({
+              school: string()
+                .min(1, 'Select a school')
+                .required('Required'),
+              degree: string().required('Required'),
+              fieldOfStudy: string().required('Required'),
+              graduationYear: number()
+                .required('Required')
+                .min(2018)
```
Comment (krubenok) [inline src/features/Application/validationSchema.ts]: Should these also be abstracted to the consts file?

Also, 2018 should be updated to 2019.

File: src/features/Application/validationSchema.ts [hunk 7]
```diff
@@ -1,80 +1,262 @@
...
+                .max(2025),
```
Comment (krubenok) [inline src/features/Application/validationSchema.ts]: Ditto.

File: src/features/Application/validationSchema.ts [hunk 8]
```diff
@@ -1,80 +1,262 @@
...
```
Comment (krubenok) [inline src/features/Application/validationSchema.ts]: Ditto.

File: src/features/Application/validationSchema.ts [hunk 9]
```diff
@@ -1,80 +1,262 @@
...
```
Comment (krubenok) [inline src/features/Application/validationSchema.ts]: ditto.

File: src/features/Application/validationSchema.ts [hunk 10]
```diff
@@ -1,80 +1,262 @@
...
+                .test(
+                  'true',
+                  'You must accept the MLH policies',
+                  (value) => value
+                ),
+              codeOfConduct: boolean()
+                .required('Required')
+                .test(
+                  'true',
+                  'You must accept the McHacks policies',
```
Comment (krubenok) [inline src/features/Application/validationSchema.ts]: Can we abstract the hackathon name to the constants file here?

Comment (loreina) [reply src/features/Application/validationSchema.ts]: will be updating Qs and error msgs in later PR

File: src/features/Application/ManageApplicationContainer.tsx (hunk 1 above)
Comment (loreina) [reply src/features/Application/ManageApplicationContainer.tsx]: different task already made

**Pull request merged**